src/jvmMain/kotlin/com/charleskorn/kaml/JvmYamlReading.kt

```diff
@@ -0,0 +1,62 @@
+/*
+
+   Copyright 2018-2023 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+package com.charleskorn.kaml
+
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.serializer
+import okio.source
+import java.io.InputStream
+
+/**
+ * Decode a YAML value [T] from an [InputStream].
+ *
+ * [InputStream] must be encoded with UTF-8.
+ */
+// The character encoding is not configurable, because we use Okio which doesn't support converting
+// between UTF-8 and other encodings.
+public fun <T> Yaml.decodeFromStream(
+    deserializer: DeserializationStrategy<T>,
+    source: InputStream,
+): T = decodeFromSource(
+    deserializer = deserializer,
+    source = source.source(),
+)
```

Comment:
* Personal preference,
* adding explicit names helped when I was learning about the code (especially when there are overloaded functions),
* in an IDE I like that I can cmd+click on a parameter name to go straight to it,
* and when I was refactoring so that when I modified the args the compilation errors were more explicit.

I'd be happy to remove them, if you prefer!

Reply:
Happy to keep them.